Опциональный проект Outsmart в desktop v 0.4

kirill-io opened the pull request.

kirill-io committed:
feat: Добавлена папка image, заполнен head

Добавил в корень проекта папку image, где будут хранится изображения использованные в проекте.
Дабавлена и подключена favicon.
Заполнен блок head в index.html

kirill-io committed:
feat: Добавлен normalize

В проект добавлен и подключен normalize.css

kirill-io committed:
feat: Добавлены шрифты

Добавлены и подключены шрифты к проету

kirill-io committed:
feat: добавлена секция promo в desktop версии

Сверстана секция promo в desktop версии
Добавлены стили по состоянию у элементов <a> и <button>

kirill-io committed:
feat: добавлена секция coverage в desktop версии

Сверстана секция coverage в desktop версии

kirill-io committed:
feat: добавлена секция services в desktop версии

Сверстана секция services в desktop версии
Добавлены состояния по hover

kirill-io committed:
feat: добавлена секция approach в desktop версии

Сверстана секция approach в desktop версии

kirill-io committed:
feat: добавлена секция certifications в desktop версии

Сверстана секция certifications в desktop версии

kirill-io committed:
feat: добавлена секция afterword в desktop версии

Сверстана секция afterword в desktop версии

kirill-io committed:
Добавлена секция outsmart в desktop версии

Сверстана секция outsmart в desktop версии

kirill-io committed:
feat: Добавлены секции Portfolio, Customers, Touch, Footer в desktop версии

Сверстаны секции Portfolio, Customers, Touch, Footer в desktop версии

kirill-io committed:
fix: Исправлена ошибка yarn

Исправлена ошибка связанная с неправильной маршрутизацией css-файлов и запущен yarn

kirill-io committed:
refactor: Произведен рефакторонг кода в index.html

Тег div секции promo заменен на тег section. А также в секции promo поменял местами блоки promo__content и header, чтобы исправить предупреждение валидатора.
Тег aside секции coverage изменен на тег section.
Тег article секции afterword изменен на тег section

kirill-io committed:
style: Изменены стили блока promo__container

После рефакторинга кода в секции promo изменены стили блока promo__container для соответствия с макетом

kirill-io committed:
refactor: Произведен рефакторонг кода в index.html

Реорганизованы классы у тегов section в блоки wrapper с соответствующими модификаторами.
Реорганизованы блоки container, добавлены соответствующие модификаторы.
Модифицированы файлы в директории blocks в соответствии с реорганизацией классов

kirill-io committed:
fix: Исправлена секция promo.

Overlay секции promo исправлен, подключен инлайноми и используется векторное изображение.
Логотип сайта реализован без ссылки

kirill-io committed:
fix: Исправлена секция coverage.

Overlay секции coverage исправлен, подключен инлайноми и используется векторное изображение

kirill-io committed:
refactor: Поправлен отступ в карточке Product developers секции coverage

Добавлен отсутствующий div с классом coverage__content в карточку Product developers

kirill-io committed:
style: Поправлена стилизация input в форме при focus.
>>
>> Свойствo border при focus на input заменено на outline, добавлено свойство box-shadow.
Заменил png-иконку close-icon на svg-иконку

kirill-io committed:
fix: Произведены изменения в секции promo

Блок logo подключен к странице через тег img.
Overlay добавлен в формате svg и подключен через тег img.
Переделаны кнопки в секции promo, а именно обернуты в div.
Подготовлены и подключены изображения в формате webp.
Подготовлены и подключены изображения для Retina-дисплеев

kirill-io committed:
style: В секцию coverage добавлены изображения webp и x2 версия

В секцию coverage добавлены и подключены фоновые изображения в формате webp и x2 версия для Retina дисплеев

kirill-io committed:
style: В секцию services добавлены изображения webp и x2 версия

В секцию services добавлены и подключены фоновые изображения в формате webp и x2 версия для Retina дисплеев

kirill-io committed:
fix: Переделана секция approach

В изображениях фоновое изображение добавлено фоном, svg изображение добавлено на страницу через тег img. Также добавлен формат webp для фоновых изображений и реализована x2 версия для Retina-дисплеев.
Блок numbered-cards переделан с grid позиционирования на flex позиционирование

kirill-io committed:
fix: Убран горизонтальный скролл в секциях promo и coverage.

Убран появляющийся горизонтальный скролл в секциях promo и coverage

kirill-io committed:
fix: Переделана секция coverage

Блок coverage переделан с grid позиционирования на flex позиционирование

kirill-io committed:
fix: Изменены логотипы секции certifications

Логотипы секции certifications добавлены в виде svg-файлов

kirill-io committed:
fix: Изменено изображение секции afterword

Для изображения секции afterword подключен формат webp и реализован вариант для Retina-дисплеев

kirill-io committed:
fix: Изменена секция outsmart

В секции outsmart переделано позиционирование с grid на flex.
Изобрабражения добавлены в формате svg

kirill-io committed:
fix: Изменена секция portfolio

Изображения в секции добавлены в формате svg.
Изменена позиция логотипов, логотипы подняты на 5px

kirill-io committed:
fix: Изменена секция customers.

Изображения в секции добавлены в формате svg.
Позиционирование элементов изменено с grid на flex

kirill-io committed:
fix: Изменена секция touch.

Фоновое изображение добавлено в формате webp и для retina дисплеев

kirill-io committed:
fix: Изменена секция footer.

В контент блока footer__copyright добавлен год.
В блок footer__row-bottom добавлен верхний отступ

kirill-io committed:
fix: Из проекта удален normalize

kirill-io committed:
feat: Сборка проекта

kirill-io committed:
fix: Исправлены секции Promo, Services, Outsmart, Touch

В секции Promo кнопки меню заменены на тег button, svg изобрание подключено атрибутом href.
В секции Services убран cursor: pointer.
В секции Outsmart использован корректный отступ между иконками и текстом.
В секции Touch требуемые состояния полей реализованы через :invalid, поправлен шрифт, у кнопки Send message добавлены состояния по ховеру, в поле Message реализовано увеличение его по вертикали при увеличении контента

kirill-io committed:
feat: Сборка проекта

kirill-io committed:
fix: В проект внесены правки

Секция Promo изменена высота секции, высота равняется размеру экрана.
Добавлен border элементу textarea.
Скрипт изменения высоты элемента textarea перенесен из html в отдельный файл

janguardian reviewed (commented):
PR одобрен. Дополнительные комментарии:

**Масштабирование**

Первый оверлей должен растягиваться на всю высоту видимой области экрана. Сейчас он не превышает 920 пикселей.
При изменении размера по вертикали полю message следует задать такой же нижний паддинг, как и верхний.
Поле message не должно растягиваться мышкой по горизонтали.

**Семантика**

Разделитель по семантике должен быть выполнен через <hr>

**Графика**

Экспортированные изображения не должны быть пережаты или иным образом обработаны так, чтобы у них присутствовали нежелательные визуальные артефакты — напиши мне в личку, проведу мастер-класс по выгрузке в webp и png

kirill-io committed:
fix: В проект внесены измнения.
>>
>> Overlay секции promo растягивается на всю ширину секции promo.
>> Исправлен padding, убрана возможность растяжения поля message в секции touch.
>> Разделитель между секциями outsmart и portfolio выполнен тегом <hr>.
>> Изображения сжаты без потерь визуального качества, для background'ов в проекте используются по одному виду изображений (убраны изображения direct и reverse)

kirill-io committed:
fix: Исправлено поле message в секции Touch.

В секции Touch у поля message исключена возможность его рассширения, поправлен начальный размер по высоте поля
